pkg/doc/verifiable/linked_data_proof.go

```diff
@@ -38,7 +37,7 @@ type verifierSignatureSuite interface {
 	signatureSuite
 
 	// Verify will verify signature against public key
-	Verify(pubKey []byte, doc []byte, signature []byte) error
+	Verify(pubKey *verifier.PublicKey, doc []byte, signature []byte) error
```

Comment:
add a TODO to replace `verifier.PublicKey` with `keyset.Handle`.

`Verify()` implementations should call pkg/crypto's Verify() function with the instance of keyset.Handle passed here.

And the same thing for Sign() implementations, they should use pkg/crypto's default Sign() function that accepts a keyset.Handle as argument.

Reply:
I think we wanted to don't make tight coupling with crypto/kms? Just make generic interfaces of signer / verifier without passing a private key as an argument to signer? @troyronda

Reply:
or if we don't want to limit our APIs to Tink, then make the pubKey argument to be `interface{}` type.

and default implementation will use `*keyset.Handle` Tink type

other implementations requiring []byte can use `*verifier.PublicKey`

Reply:
> I think we wanted to don't make tight coupling with crypto/kms? Just make generic interfaces of signer / verifier without passing a private key as an argument to signer? @troyronda

the concept of hiding keys entails using a key reference to execute a crypto primitive, which should be resolved by a kms implementation and primitive executed by a crypto implementation.

in order to not tie the code to any specific implementation, we should simply use `interface{}` type for keys. This way, kms/crypto Tink implementations can exchange `*keyset.Handle` with verifierSignatureSuite/signerSignatureSuite implemented with Tink key handles. Other KMS implementations will exchange their own key type with their respective signature suite implementations.

Reply:
I see your point. Thank you for the explanation! I created https://github.com/hyperledger/aries-framework-go/issues/1458.